src/runner.ts

```diff
@@ -183,6 +185,248 @@ export class CoverageParserRunner {
         });
     }
 
+    private mergeCoberturaReports(reportPaths: string[]): types.CoberturaCoverage | undefined {
+        if (!reportPaths.length) {
+            return undefined;
+        }
+
+        let baseReportPath: string = reportPaths[0];
+        core.debug(messagesFormatter.format(messages.using_cobertura_report_as_base_report, baseReportPath));
+
+        let baseCoverage = this.processXMLToObj(baseReportPath);
+        if (reportPaths.length > 1) {
+            for (let i = 1; i < reportPaths.length; i++) {
+                const reportToMerge: types.CoberturaCoverage = this.processXMLToObj(reportPaths[i]);
+                try {
+                    core.debug(messagesFormatter.format(messages.merging_cobertura_report, reportPaths[i]));
+                    baseCoverage = this.mergeCoberturaCoverage(lodash.cloneDeep(baseCoverage), reportToMerge);
+                } catch (error) {
+                    if (error instanceof Error) {
+                        core.warning(messagesFormatter.format(messages.coverage_data_was_not_merged_due_to, reportPaths[i], error.message));
+                    } else {
+                        core.warning(messagesFormatter.format(messages.coverage_data_was_not_merged, reportPaths[i])); // Should never happen
+                    }
+                }
+            }
+
+            this.updateAttributes(baseCoverage);
+        }
+
+        fs.writeFileSync(this.MERGED_COBERTURA_REPORT_PATH, this.processObjToXML(baseCoverage), 'utf-8');
```

Comment:
line215: This code generates the merged Cobertura report as an XML file. It can be removed if there is no requirement to provide the report for user review or upload it to artifacts.

Reply:
I lean to keep this merged file as it provides users with some information to debug with before they contact us if anything happens. @rwang-parasoft What do you think? 
Additionally, if we keep it, I think we should try to upload the merged report as artifact as default behavior if possible. If that's not possible, at least we should print the path of the merged report for reference.
I can double check with Nathan in tomorrow's sync up meeting if needed.

Reply:
I think we can keep this file, and print the path of the merged report for reference. No need to upload it to artifact currently. @yshi-parasoft you can double check with Nathan.

Reply:
Feedback: let's delete the merged report and related logic.